### ISSUE TITLE
Error Type: oneflow.ErrorProto.op_kernel_not_found_error

### ISSUE THREAD

**Issue by Jeayea** [Oneflow-Inc/oneflow#9341]
1. 问题描述：
用oneflow加载torch模型，在同样的conda 环境下，偶尔能成功运行，大部分时间显示 Cannot find the kernel matching Current OperatorConf.
2. 报错信息：
`F20221031 02:22:26.948527 111108 exec_graph.cpp:121] 
  File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/graph/exec_graph.cpp", line 121, in InferBlobDescs
    op_->InferBlobDescsIf(GetBlobDesc4BnInOp, parallel_ctx ... oneflow/nn/modules/sparse.py', line 166, code forward>;
  File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/operator/operator.cpp", line 328, in InferBlobDescsIf
    InferInternalBlobDescsIf(GetBlobDesc4BnInOp, parallel_ctx, job_desc)
  File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/operator/user_op.cpp", line 544, in InferInternalBlobDescs
    user_op::UserOpRegistryMgr::Get().GetOpKernelRegistryResult( op_conf().user_conf().op_type_name(), UserOpKernelRegContext(this, GetBlobDesc4BnInOp, parallel_ctx))
Cannot find the kernel matching Current OperatorConf. 
 The Info of OperatorConf are 
 op_name: decoder.embedding-embedding-7887
 op_type_name: embedding
 DeviceType_Name: kCPU
 DataType_Name of weight_0: kFloat
 DataType_Name of indices_0: kFloat
 DataType_Name of out_0: kFloat
Error Type: oneflow.ErrorProto.op_kernel_not_found_error
*** Check failure stack trace: ***
    @     0x7f1057e2e62a  google::LogMessage::Fail()
    @     0x7f1057e2e912  google::LogMessage::SendToLog()
    @     0x7f1057e2e197  google::LogMessage::Flush()
    @     0x7f1057e30d09  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f105135ab8c  oneflow::ExecNode::InferBlobDescs()
    @     0x7f105135cf75  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8ExecNodeEEZZNKS0_5GraphIS3_NS0_8ExecEdgeEE15TopoForEachNodeESt8functionIFvS4_EEENKUlPKcE_clESD_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7f105135e538  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f10513d8887  oneflow::NormalForwardCompTaskNode::BuildExecGphAndRegst()
    @     0x7f10513a6b45  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8TaskNodeEEZZNKS0_5GraphIS3_NS0_8TaskEdgeEE15TopoForEachNodeESt8functionIFvS4_EEENKUlPKcE_clESD_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7f10513a6f98  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f105142d193  oneflow::Compiler::Compile()
    @     0x7f1050e08ea5  oneflow::NNGraph::CompileAndInitRuntime()
    @     0x7f10b5e6f7fc  (unknown)
    @     0x7f10b6029329  (unknown)
    @           0x5f6929  PyCFunction_Call
    @           0x5f74f6  _PyObject_MakeTpCall
    @           0x50c383  (unknown)
    @           0x570b26  _PyEval_EvalFrameDefault
    @           0x50bb2e  (unknown)
    @           0x570b26  _PyEval_EvalFrameDefault
    @           0x569dba  _PyEval_EvalCodeWithName
    @           0x5f6eb3  _PyFunction_Vectorcall
    @           0x50bc2c  (unknown)
    @           0x5f6082  PyObject_Call
    @           0x56d2d5  _PyEval_EvalFrameDefault
    @           0x569dba  _PyEval_EvalCodeWithName
    @           0x5f6eb3  _PyFunction_Vectorcall
    @           0x59d81e  (unknown)
    @           0x5f74f6  _PyObject_MakeTpCall
    @           0x570d55  _PyEval_EvalFrameDefault
    @           0x569dba  _PyEval_EvalCodeWithName
    @           0x5f6eb3  _PyFunction_Vectorcall
Aborted (core dumped)`

3. 代码： 
https://colab.research.google.com/drive/1pfTw3R5zLWPGO_SheVuWBR2QL0Y0l_s8?usp=sharing
4. torch模型的权重：
https://drive.google.com/file/d/1-cOi4su6vU-MRHiJMXVMtnoyeHhK1w9g/view?usp=share_link
https://drive.google.com/file/d/1-_2AugQqonm9vUb7z8cNO0ij1F8UuaH4/view?usp=share_link

5. 运行环境：
Package                 Version             
absl-py                 1.3.0               
antlr4-python3-runtime  4.8                 
apturl                  0.5.2               
bcrypt                  3.1.7               
bitarray                2.6.0               
blinker                 1.4                 
Brlapi                  0.7.0               
certifi                 2019.11.28          
cffi                    1.15.1              
chardet                 3.0.4               
Click                   7.0                 
colorama                0.4.3               
command-not-found       0.3                 
commonmark              0.9.1               
contourpy               1.0.5               
cryptography            2.8                 
cupshelpers             1.0                 
cycler                  0.11.0              
Cython                  0.29.32             
dbus-python             1.2.16              
defer                   1.0.6               
distro                  1.4.0               
distro-info             0.23ubuntu1         
duplicity               0.8.12.0            
entrypoints             0.3                 
fairseq                 0.12.2              
fasteners               0.14.1              
filelock                3.8.0               
fonttools               4.37.4              
future                  0.18.2              
httplib2                0.14.0              
huggingface-hub         0.10.1              
hydra-core              1.0.7               
idna                    2.8                 
importlib-resources     5.10.0              
joblib                  1.2.0               
keyring                 18.0.1              
kiwisolver              1.4.4               
language-selector       0.1                 
launchpadlib            1.10.13             
lazr.restfulclient      0.14.2              
lazr.uri                1.0.3               
lockfile                0.12.2              
louis                   3.12.0              
lxml                    4.9.1               
macaroonbakery          1.3.1               
Mako                    1.1.0               
MarkupSafe              1.1.0               
matplotlib              3.6.1               
monotonic               1.5                 
netifaces               0.10.4              
nltk                    3.7                 
numpy                   1.23.4              
oauthlib                3.1.0               
olefile                 0.46                
omegaconf               2.0.6               
oneflow                 0.8.0               
packaging               21.3                
pandas                  1.5.1               
paramiko                2.6.0               
pexpect                 4.6.0               
Pillow                  7.0.0               
pip                     20.0.2              
portalocker             2.6.0               
protobuf                3.20.3              
pycairo                 1.16.2              
pycparser               2.21                
pycups                  1.9.73              
Pygments                2.13.0              
PyGObject               3.36.0              
PyJWT                   1.7.1               
pymacaroons             0.13.0              
PyNaCl                  1.3.0               
pyparsing               3.0.9               
pyRFC3339               1.1                 
python-apt              2.0.0+ubuntu0.20.4.7
python-dateutil         2.8.2               
python-debian           0.1.36ubuntu1       
pytz                    2022.5              
pyxdg                   0.26                
PyYAML                  5.3.1               
regex                   2022.9.13           
reportlab               3.5.34              
requests                2.22.0              
requests-unixsocket     0.2.0               
rich                    12.6.0              
rouge-score             0.1.2               
sacrebleu               2.3.1               
screen-resolution-extra 0.0.0               
SecretStorage           2.3.1               
setuptools              45.2.0              
simplejson              3.16.0              
six                     1.14.0              
ssh-import-id           5.10                
systemd-python          234                 
tabulate                0.9.0               
timm                    0.6.11              
torch                   1.12.1+cu116        
torchaudio              0.12.1+cu113        
torchvision             0.13.1+cu113        
tqdm                    4.64.1              
typing-extensions       4.4.0               
ubuntu-advantage-tools  27.6                
ubuntu-drivers-common   0.0.0               
ufw                     0.36                
unattended-upgrades     0.1                 
urllib3                 1.25.8              
usb-creator             0.3.7               
wadllib                 1.3.3               
wheel                   0.34.2              
xkit                    0.0.0               
zipp                    3.10.0

**Comment by Jeayea** [comment]
```
from __future__ import unicode_literals, print_function, division
from io import open
import unicodedata
import string
import re
import random
import os
os.system("wget https://raw.githubusercontent.com/sunnysai12345/News_Summary/master/news_summary_more.csv")
import pandas as pd
data_path = "./news_summary_more.csv"
data = pd.read_csv(data_path,encoding="utf-8")

import nltk 
nltk.download("stopwords")
import re
from nltk.corpus import stopwords

stop_words = stopwords.words("english")

def preprocess(text):
    text = text.lower() # lowercase
    text = text.split() # convert have'nt -> have not
    for i in range(len(text)):
        word = text[i]
    text = " ".join(text)
    text = text.split()
    newtext = []
    for word in text:
        if word not in stop_words:
            newtext.append(word)
    text = " ".join(newtext)
    text = text.replace("'s","") # convert your's -> your
    text = re.sub(r"\(.*\)","",text) # remove (words)
    text = re.sub(r"[^a-zA-Z0-9. ]","",text) # remove punctuations
    text = re.sub("\."," . ",text)
    return text

data['headlines'] = data['headlines'].apply(lambda x:preprocess(x))
data['text'] = data['text'].apply(lambda x:preprocess(x))

x = data['text']
y = data['headlines']


import oneflow
from oneflow import optim
import oneflow.nn as nn
import oneflow.nn.functional as F
device = oneflow.device("cpu")

SOS_token = 0
EOS_token = 1


class Lang:
    def __init__(self, name):
        self.name = name
        self.word2index = {}
        self.word2count = {}
        self.index2word = {0: "SOS", 1: "EOS"} # add PAD 
        self.n_words = 2  # Count SOS and EOS and PAD

    def addSentence(self, sentence):
        for word in sentence.split(' '):
            self.addWord(word)

    def addWord(self, word):
        if word not in self.word2index:
            self.word2index[word] = self.n_words
            self.word2count[word] = 1
            self.index2word[self.n_words] = word
            self.n_words += 1
        else:
            self.word2count[word] += 1

def readLangs(text, summary, reverse=False):
    print("Reading lines...")
    
    # Split every line into pairs and normalize
    pairs = [[text[i],summary[i]] for i in range(len(text))]

    # Reverse pairs, make Lang instances
    if reverse:
        pairs = [list(reversed(p)) for p in pairs]
        input_lang = Lang(summary)
        output_lang = Lang(text)
    else:
        input_lang = Lang(text)
        output_lang = Lang(summary)

    return input_lang, output_lang, pairs

def prepareData(lang1, lang2, reverse=False):
    input_lang, output_lang, pairs = readLangs(lang1, lang2, reverse)
    print("Read %s sentence pairs" % len(pairs))
    print("Counting words...")
    for pair in pairs:
        input_lang.addSentence(pair[0])
        output_lang.addSentence(pair[1])
    print("Counted words:")
    print(input_lang.name, input_lang.n_words)
    print(output_lang.name, output_lang.n_words)
    return input_lang, output_lang, pairs

input_lang, output_lang, pairs = prepareData( x, y , False)

class EncoderRNN(nn.Module):
    def __init__(self, input_size, hidden_size):
        super(EncoderRNN, self).__init__()
        self.hidden_size = hidden_size

        self.embedding = nn.Embedding(input_size, hidden_size,  padding_idx=0)
        self.gru = nn.GRU(hidden_size, hidden_size)

    def forward(self, input, hidden):
        embedded = self.embedding(input).view(1, 1, -1)
        output = embedded
        output, hidden = self.gru(output, hidden)
        return output, hidden

    def initHidden(self):
        return oneflow.zeros(1, 1, self.hidden_size, device=device)

MAX_LENGTH = 90

class AttnDecoderRNN(nn.Module):
    def __init__(self, hidden_size, output_size, dropout_p=0.1, max_length=MAX_LENGTH):
        super(AttnDecoderRNN, self).__init__()
        self.hidden_size = hidden_size
        self.output_size = output_size
        self.dropout_p = dropout_p
        self.max_length = max_length
        self.embedding = nn.Embedding(self.output_size, self.hidden_size, padding_idx=0) # need to be update
        self.attn = nn.Linear(self.hidden_size * 2, self.max_length)
        self.attn_combine = nn.Linear(self.hidden_size * 2, self.hidden_size)
        self.dropout = nn.Dropout(self.dropout_p)
        self.gru = nn.GRU(self.hidden_size, self.hidden_size)
        self.out = nn.Linear(self.hidden_size, self.output_size)

    def forward(self, input, hidden, encoder_outputs):
        embedded = self.embedding(input)
        embedded = embedded.view(-1, 1, self.hidden_size)
        embedded = self.dropout(embedded)
        attn_weights = F.softmax(
            self.attn(oneflow.cat((embedded[0], hidden[0]), 1)), dim=1) 
        attn_applied = oneflow.bmm(attn_weights.unsqueeze(0),
                                 encoder_outputs.unsqueeze(0))

        output = oneflow.cat((embedded[0], attn_applied[0]), 1)
        output = self.attn_combine(output).unsqueeze(0)

        output = F.relu(output)
        output, hidden = self.gru(output, hidden)

        output = F.log_softmax(self.out(output[0]), dim=1)
        return output, hidden, attn_weights

    def initHidden(self):
        return oneflow.zeros(1, 1, self.hidden_size, device=device)

def indexesFromSentence(lang, sentence, max_length = MAX_LENGTH-1):
    return [lang.word2index[word] for word in sentence.split(' ')]

def tensorFromSentence(lang, sentence):
    indexes = indexesFromSentence(lang, sentence)
    indexes.append(EOS_token)
    return oneflow.tensor(indexes, dtype=oneflow.long, device=device).view(-1, 1)
def tensorsFromPair(pair):
    input_tensor = tensorFromSentence(input_lang, pair[0])
    target_tensor = tensorFromSentence(output_lang, pair[1])
    return (input_tensor, target_tensor)

class MyGraph(nn.Graph):
    def __init__(self, encoder, decoder, max_length=MAX_LENGTH):
        super().__init__()
        self.encoder = encoder
        self.decoder = decoder
        self.max_length = max_length
        
    def build(self, input_tensor):
        with oneflow.no_grad():
            input_length = input_tensor.size()[0]
            encoder_hidden = self.encoder.initHidden()
            encoder_outputs = oneflow.zeros(self.max_length, self.encoder.hidden_size, device=device)


            for ei in range(input_length):
                encoder_output, encoder_hidden = self.encoder(input_tensor[ei],
                                                        encoder_hidden)
                encoder_outputs[ei] += encoder_output[0, 0] # EOS

            decoder_input = oneflow.tensor([[SOS_token]], device=device)  # SOS
            decoder_hidden = encoder_hidden

            decoded_words = []
            decoder_attentions = oneflow.zeros(self.max_length, self.max_length)

            for di in range(self.max_length):
                decoder_output, decoder_hidden, decoder_attention = self.decoder(
                    decoder_input, decoder_hidden, encoder_outputs)
                decoder_attentions[di] = decoder_attention.data
                topv, topi = decoder_output.data.topk(1)
                decoded_words.append(topi)
            return decoded_words

import torch

def load_weights(model, weight_path):
  parameters = torch.load(weight_path)
  for key, value in parameters.items():
      val = value.detach().cpu().numpy()
      parameters[key] = val
  model.load_state_dict(parameters)
def evaluateTop10_graph(graph, n=10):
    for i in range(n):
        pair = pairs[i]
        print('>', pair[0])
        print('=', pair[1])
        input_tensor = tensorFromSentence(input_lang, pair[0])
        decoded_words = graph(input_tensor)
        print("finish decoding")
        output_words = []
        for topi in decoded_words:
          if topi.item() == EOS_token:
              output_words.append('<EOS>')
              break
          else:
              output_words.append(output_lang.index2word[topi.item()])

        output_sentence = ' '.join(output_words)
        print('<', output_sentence)
        print('')
enc_weight_path = './drive/MyDrive/summa/weights/enc_17500.w'
att_weight_path = './drive/MyDrive/summa/weights/att_17500.w'


hidden_size = 300
encoder1 = EncoderRNN(input_lang.n_words, hidden_size).to(device)
attn_decoder1 = AttnDecoderRNN(hidden_size, output_lang.n_words, dropout_p=0.1).to(device)

load_weights(encoder1, enc_weight_path)
load_weights(attn_decoder1, att_weight_path)

import time
mygraph = MyGraph(encoder1, attn_decoder1)
st_graph = time.time()
evaluateTop10_graph(mygraph)
ed_graph = time.time()

print("inference time using static graph:{}".format(ed_graph - st_graph))
```

**Comment by shangguanshiyuan** [comment]
你好，nn.Embedding接受整形的indices输入，此处报错是因为输入为浮点类型，未找到匹配相同类型的OpConf，请检查输入的类型是否正确。

**Comment by Jeayea** [comment]
好的好的，感谢！
